sdk/lib/cmlib/hivewrt.c

```diff
@@ -217,7 +217,7 @@ HvpWriteLog(
         /* Check if the block is clean or we're past the last block */
         LastIndex = BlockIndex;
         BlockIndex = RtlFindSetBits(&RegistryHive->DirtyVector, 1, BlockIndex);
-        if (BlockIndex == ~HV_CLEAN_BLOCK || BlockIndex < LastIndex)
+        if (BlockIndex == 0xFFFFFFFF || BlockIndex < LastIndex)
```

Comment:
RtlFindSetBits is documented to return 0xFFFFFFFF if no bits are set, that has nothing to do with HV_CLEAN_BLOCK. I can change the other two instances as well.

Reply:
It was my fault on poorly understanding the returned construct value of `RtlFindSetBits`, thinking that it would return 0 if the target hive log block is clean, apologies. If it's okay with you @ThFabba, perhaps you can make the `HV_CLEAN_BLOCK` construct to be `0xFFFFFFFF` and simply do `BlockIndex == HV_CLEAN_BLOCK` on code paths that use it.